crates/alloy/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+#![doc = include_str!("../README.md")]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/alloy-rs/core/main/assets/alloy.jpg",
+    html_favicon_url = "https://raw.githubusercontent.com/alloy-rs/core/main/assets/favicon.ico"
+)]
+#![warn(
+    missing_copy_implementations,
+    missing_debug_implementations,
+    missing_docs,
+    unreachable_pub,
+    clippy::missing_const_for_fn,
+    rustdoc::all
+)]
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+#![deny(unused_must_use, rust_2018_idioms)]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+
+// Just for features.
+#[cfg(feature = "transport-http-reqwest")]
+use reqwest as _;
+
+#[doc(inline)]
+pub use alloy_core::*;
+
+#[cfg(feature = "contract")]
+#[doc(inline)]
+pub use alloy_contract as contract;
+
+#[cfg(feature = "consensus")]
+#[doc(inline)]
+pub use alloy_consensus as consensus;
+
+#[cfg(feature = "eips")]
+#[doc(inline)]
+pub use alloy_eips as eips;
+
+#[cfg(feature = "network")]
+#[doc(inline)]
+pub use alloy_network as network;
+
+#[cfg(feature = "genesis")]
+#[doc(inline)]
+pub use alloy_genesis as genesis;
+
+#[cfg(feature = "node-bindings")]
+#[doc(inline)]
+pub use alloy_node_bindings as node_bindings;
+
+/// Interface with an Ethereum blockchain.
+///
+/// See [`alloy_providers`] for more details.
+#[cfg(feature = "providers")]
+pub mod providers {
+    #[doc(inline)]
+    pub use alloy_providers::*;
+
+    // TODO: provider type aliases
+    // #[cfg(feature = "provider-http")]
```

Comment:
tbd in this pr?

Reply:
don't really know how to hook it up, and it will change later with network abstraction